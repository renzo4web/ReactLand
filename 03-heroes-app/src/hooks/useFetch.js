import { useEffect, useState } from 'react';

export const useFetch = ({ name }) => {
  const [hero, setHero] = useState({});

  const tk = '301715191729878';

  useEffect(() => {
    let isCurrent = true;

    fetch('https://superheroapi.com/api/301715191729878/search/superman', {
      method: 'GET',
      headers: {
        'Access-Control-Allow-Origin': '*',
      },
    })
      .then(response => {
        console.log(response);
      })
      .catch(err => {
        console.error(err);
      });

    return () => {
      isCurrent = false;
    };
  }, [name]);

  return hero;
};
