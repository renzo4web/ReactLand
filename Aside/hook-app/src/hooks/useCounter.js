import { useState } from "react";

export const useCounter = (initialState = 10) => {
  const [state, setState] = useState(initialState);

  const increment = () => {
    setState((prev) => prev + 1);
  };
  const decrement = () => {
    setState((prev) => prev - 1);
  };
  const reset = () => {
    setState(initialState);
  };

  return {
    state,
    reset,
    increment,
    decrement,
  };
};
