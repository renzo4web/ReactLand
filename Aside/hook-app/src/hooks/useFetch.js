import { useEffect, useState, useRef } from "react";

export const useFetch = (url) => {
  const isMounted = useRef(true);

  const [data, setData] = useState({
    text: null,
    loading: true,
    error: null,
  });

  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);

  useEffect(() => {
    setData({ ...data, loading: true });
    fetch(url)
      .then((data) => data.json())
      .then((quote) => {
        if (isMounted.current) {
          setData({ ...data, text: quote[0], loading: false });
        } else {
          console.log("El State no se monto");
        }
      })
      .catch((err) => setData({ ...data, loading: false, error: err }));
  }, [url]);

  return data;
};
