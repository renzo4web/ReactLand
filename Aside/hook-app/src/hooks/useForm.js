import React, { useState } from "react";

export const useForm = (formInputs = {}) => {
  const [inputs, setInputs] = useState(formInputs);

  const reset = () => {
    setInputs(formInputs);
  };

  const handleChange = ({ target }) => {
    const { name, value } = target;

    setInputs({
      ...inputs,
      [name]: value,
    });
  };
  return [inputs, handleChange, reset];
};
