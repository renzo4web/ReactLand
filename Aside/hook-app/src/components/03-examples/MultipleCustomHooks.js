import React, { useEffect, useState } from "react";
import { useFetch } from "../../hooks/useFetch";

const MultipleCustomHooks = () => {
  const [id, setId] = useState(1);
  const url = `https://www.breakingbadapi.com/api/quotes/${id}`;
  const [data, setEndpoint] = useFetch(url);
  const { loading, text, error } = data;
  console.log(data);
  return (
    <div>
      {loading ? (
        <h4>Loading</h4>
      ) : (
        <article>
          <h4>{text.series}</h4>
          <p>{text.quote}</p>
          <button
            onClick={() => {
              setId((prev) => prev + 1);
              console.log(id);
              console.log(url);
              setEndpoint(url);
            }}
          >
            Next quote
          </button>
        </article>
      )}
    </div>
  );
};

export default MultipleCustomHooks;
