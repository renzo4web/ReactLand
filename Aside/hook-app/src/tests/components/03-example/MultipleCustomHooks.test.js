import { shallow } from "enzyme";
import React from "react";
import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
jest.mock("../../../hooks/useFetch");

describe("<MultipleCustomHooks /> test", () => {
  test("should render properly ", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should  render the information", () => {
    useFetch.mockReturnValue({
      text: {
        series: "Breaking Bad",
        quote: "",
        author: "gol",
      },
      loading: false,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find("h4").text().trim()).toBe(
      "Breaking Bad"
    );
  });
});
