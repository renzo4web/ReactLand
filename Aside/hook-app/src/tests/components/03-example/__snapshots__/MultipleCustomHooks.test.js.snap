// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleCustomHooks /> test should render properly  1`] = `
<div
  className="bg-light p-3"
>
  <h4
    className="alert alert-info"
  >
    Loading
  </h4>
</div>
`;
